source/base/utilities.cc

```diff
@@ -132,7 +132,7 @@ namespace Utilities
   trim(const std::string &input)
   {
     std::string::size_type left = 0;
-    std::string::size_type right = input.size() - 1;
+    std::string::size_type right = input.size() > 0 ? input.size() - 1 : 0;
```

Comment:
Two surprises:
1. such a test already exists
2. Everything seems fine (no complaints in GDB or valgrind) if one accesses the `std::string::npos` (a.k.a. `-1`) entry of an empty string: it just returns the NUL.